plugin/schema.go

```diff
@@ -0,0 +1,82 @@
+package plugin
+
+import (
+	"reflect"
+	"strings"
+)
+
+// EntrySchema represents an entry's schema.
+type EntrySchema struct {
+	Type      string        `json:"type"`
+	ShortType string        `json:"short_type"`
+	Singleton bool          `json:"singleton"`
```

Comment:
Singleton is important for `stree`. Non-singleton entries are displayed as `[<type/short_type>]`, while singleton entries are displayed as `<type/short_type>`. This way, people won't get confused into thinking there's more than one "resources" directory, or "metadata.json" file.

`Type` is meant to be the raw (language-specific) class-type. Don't really know when/if that will be used by clients (here it's used to handle cycles when constructing entry schemas). `ShortType` is important so that you avoid printing something like `ec2Dir/ec2InstancesDir/ec2instance/ec2InstanceMetadataJSON` which doesn't add any clarity.

(This is also one nice thing about the `*Template()` functions because if you're re-using a prior entry's template, you can override its short type to something else. Doing this with struct tags is difficult).

Reply:
Am open to a better name than `ShortType`.

Reply:
This mostly seems like it should be inline documentation.

Reply:
short type seems more akin to a description. What's the expectation for how it's used?

Reply:
Maybe `singleton` could be inline, but then a Wash user would have to figure that out themselves.

The intended use for `singleton` and `short_type`. For example
```
stree mnt/docker
mnt/docker
  containers
    [container]
      metadata.json
      log
      fs
        [dir]
          [dir]
          [file]
        [file]
  volumes
    [volume]
      [dir]
        [dir]
        [file]
      [file]
```

(`short_type` would be what's printed as "file"/"dir"/"volumes"/"container"; singleton entries don't have the `[]` around them).